c7n/resources/vpc.py

```diff
@@ -215,11 +216,11 @@ def process(self, resources, event=None):
         fattrs = list(sorted(self.attrs.intersection(self.data.keys())))
         self.ports = 'Ports' in self.data and self.data['Ports'] or ()
         self.only_ports = 'OnlyPorts' in self.data and self.data['OnlyPorts'] or ()
+        self.max_cidr_size = 'IpRanges' in self.data and self.data['IpRanges']['value'] or ()
         for f in fattrs:
             fv = self.data.get(f)
             if isinstance(fv, dict):
-                if 'key' not in fv:
-                    fv['key'] = f
+                fv['key'] = f
```

Comment:
so this is a mistake on my part, the base class here has a controlled vocabulary for validation and to change some specified keys into value filters, either we should do away with that, or parse jmespath expressions for the   first path element in an expression to validate (ie. IpRanges[].CidrIp validates against IpRanges.)